client/components/tinymce/plugins/simple-payments/dialog/form.jsx

```diff
@@ -19,121 +23,128 @@ import CompactFormToggle from 'components/forms/form-toggle/compact';
+		const { translate, currencyDefaults } = this.props;
 
 		return (
 			<form className="editor-simple-payments-modal__form">
-				<UploadImage
-					defaultImage={ fieldValues.featuredImageId }
+				<Field
+					name="featuredImageId"
 					onError={ this.handleUploadImageError }
-					onUploadImageDone={ onUploadImageDone }
+					component={ renderUpload }
 				/>
 				<div className="editor-simple-payments-modal__form-fields">
-					<FormFieldset>
-						<FormLabel htmlFor="title">
-							{ translate( 'What are you selling?' ) }
-						</FormLabel>
-						<FormTextInput
-							name="title"
-							id="title"
-							placeholder={ translate( 'Product name' ) }
-							value={ fieldValues.title }
-							onChange={ this.handleFieldChange }
-							isError={ isTitleInvalid }
-						/>
-						{ isTitleInvalid &&
-							<FormInputValidation
-								isError
-								text={ translate( 'Product name cannot be empty.' ) }
-							/> }
-					</FormFieldset>
-					<FormFieldset>
-						<FormLabel htmlFor="description">
-							{ translate( 'Description' ) }
-						</FormLabel>
-						<FormTextarea
-							name="description"
-							id="description"
-							value={ fieldValues.description }
-							onChange={ this.handleFieldChange }
-						/>
-					</FormFieldset>
-					<FormFieldset>
-						<FormLabel htmlFor="price">
-							{ translate( 'Price' ) }
-						</FormLabel>
-						<FormCurrencyInput
-							name="price"
-							id="price"
-							currencySymbolPrefix={ currencyDefaults.symbol }
-							placeholder="0.00"
-							value={ fieldValues.price }
-							onChange={ this.handleFieldChange }
-							isError={ isPriceInvalid }
-						/>
-						{ isPriceInvalid &&
-							<FormInputValidation isError text={ translate( 'Invalid price' ) } /> }
-					</FormFieldset>
-					<FormFieldset>
-						<CompactFormToggle
-							name="multiple"
-							id="multiple"
-							checked={ !! fieldValues.multiple }
-							onChange={ this.handleMultipleCheckboxChange }
-						>
-							{ translate( 'Allow people to buy more than one item at a time.' ) }
-						</CompactFormToggle>
-					</FormFieldset>
-					<FormFieldset>
-						<FormLabel htmlFor="email">
-							{ translate( 'Email' ) }
-						</FormLabel>
-						<FormTextInput
-							name="email"
-							id="email"
-							value={ fieldValues.email }
-							onChange={ this.handleFieldChange }
-							isError={ isEmailInvalid }
-						/>
-						{ isEmailInvalid &&
-							<FormInputValidation isError text={ translate( 'Invalid email' ) } /> }
-						<FormSettingExplanation>
-							{ translate(
-								'This is where PayPal will send your money.' +
-									" To claim a payment, you'll need a PayPal account connected to a bank account.",
-							) }
-						</FormSettingExplanation>
-					</FormFieldset>
+					<Field
+						name="title"
+						label={ translate( 'What are you selling?' ) }
+						placeholder={ translate( 'Product name' ) }
+						component={ renderField( FormTextInput ) }
```

Comment:
That's because `isError = meta.touched && meta.error` in `renderField`, and `meta.error` is a string. One `!!` will fix that. Good catch!